cleverhans/dataset.py

```diff
@@ -0,0 +1,146 @@
+"""Dataset class for CleverHans
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from cleverhans.utils_mnist import data_mnist
+
+keras = None  # Only load keras if user tries to use a dataset that requires it
+
+
+class Dataset(object):
+    """Abstract base class representing a dataset.
+    """
+
+    # The number of classes in the dataset. Should be specified by subclasses.
+    nb_classes = None
+
+    def get_factory(self):
+        """Returns a picklable callable that recreates the dataset.
+        """
+
+        if hasattr(self, 'args'):
```

Comment:
I would suggest not supporting args, since positional arguments can lead to errors (if the order changed in the class). On the other hand, kwargs are perfect since even if the order changes or new elements are added, it will still work fine.

Reply:
Thanks, done